python/tskit/combinatorics.py

```diff
@@ -37,6 +37,30 @@
 import tskit
 
 
+def equal_chunks(lst, k):
+    """
+    Yield k successive equally sized chunks from lst of size n.
+
+    If k >= n, we return n chunks of size 1.
+
+    Otherwise, we always return k chunks. The first k - 1 chunks will
+    contain exactly n // k items, and the last chunk the remainder.
+    """
+    n = len(lst)
+    if k <= 0 or int(k) != k:
+        raise ValueError("Number of chunks must be a positive integer")
+
+    if n > 0:
+        chunk_size = max(1, n // k)
```

Comment:
Can you do this instead with `yield from itertools.islice(lst, chunk_size)`?

Reply:
Hey, good idea. No, I don't think islice will do it, the semantics of what we want are really quite specific in order to line up with the unranked trees.  numpy has a [function](https://numpy.org/doc/stable/reference/generated/numpy.array_split.html) which almost does what we want, but not quite.